File: tpv_purse_price_variable/models/pos_order_line.py
from odoo import models, fields, api


class PosOrderLine(models.Model):
    _inherit = "pos.order.line"

    tpv_purse_program_id = fields.Many2one(
        comodel_name="loyalty.program",
        string="TPV Purse Program",
        help="Programa eWallet seleccionado para una recarga de monedero con precio variable.",
    )

    @api.model
    def _load_pos_data_fields(self, config):
        fields_list = super()._load_pos_data_fields(config)
        if "tpv_purse_program_id" not in fields_list:
            fields_list.append("tpv_purse_program_id")
        return fields_list

File: tpv_purse_price_variable/static/src/app/models/pos_order_line.js
/** @odoo-module */

import { patch } from "@web/core/utils/patch";
import { PosOrderline } from "@point_of_sale/app/models/pos_order_line";

patch(PosOrderline, {
    extraFields: {
        ...(PosOrderline.extraFields || {}),
        tpv_purse_program_id: {
            model: "pos.order.line",
            name: "tpv_purse_program_id",
            relation: "loyalty.program",
            type: "many2one",
        },
    },
});

patch(PosOrderline.prototype, {
    setOptions(options) {
        if (options.eWalletGiftCardProgram) {
            this.tpv_purse_program_id = options.eWalletGiftCardProgram;
        }
        if (options.tpvPurseProgram) {
            this.tpv_purse_program_id = options.tpvPurseProgram;
        }
        return super.setOptions(...arguments);
    },
});

